crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -4680,20 +4680,11 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         }
 
         // Handle various singletons.
-        if let Type::NominalInstance(instance) = declared.inner_type() {
-            if instance
-                .class(self.db())
-                .is_known(self.db(), KnownClass::SpecialForm)
+        if let Some(name_expr) = target.as_name_expr() {
+            if let Some(special_form) =
+                SpecialFormType::try_from_file_and_name(self.db(), self.file(), &name_expr.id)
```

Comment:
We now call `SpecialFormType::try_from_file_and_name` on literally every single annotated assignment to a Name target -- that's a lot of annotated assignments. `try_from_file_and_name` should usually be just a string compare, but that can add up? Is there any way we could filter out more of the cases here and see if that helps? Like, it now either needs to be a `KnownClass::SpecialForm` or a `SubclassOf` type?